core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -80,6 +77,65 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     new File(subDir, filename)
   }
 
+  def getFileDefault(filename: String): File = {
+    _getFile(filename, localDirs)
+  }
+  var getFile = getFileDefault _
```

Comment:
OK, after doublechecking the code, I think we'd better still keep the `getFile` as `def getFile(file: String)`, and leave the status checking within the function for contention reason.